File: backend/App.Contracts/DTOs/SchoolDto.cs
using App.Domain.Entities;

namespace App.Contracts.DTOs;

public class SchoolDto
{
    // Parameterless constructor for deserialization
    public SchoolDto() { }
    
    public SchoolDto(SchoolEntity school)
    {
        Id = school.Id;
        Name = school.Name;
        ShortName = school.ShortName;
        Domain = school.Domain;
        StudentCodePattern = school.StudentCodePattern;
    }

    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string ShortName { get; set; } = string.Empty;
    public string Domain { get; set; } = string.Empty;
    public string StudentCodePattern { get; set; } = string.Empty;
    
    public static List<SchoolDto> ToDtoList(List<SchoolEntity>? entities)
    {
        return entities?.Select(e => new SchoolDto(e)).ToList() 
               ?? new List<SchoolDto>();
    }
}

File: mobile/lib/models/Responses/school_dto.dart
class SchoolDto {
  final String id;
  final String name;
  final String shortName;
  final String domain;
  final String studentCodePattern;

  SchoolDto({
    required this.id,
    required this.name,
    required this.shortName,
    required this.domain,
    required this.studentCodePattern,
  });

  factory SchoolDto.fromJson(Map<String, dynamic> json) {
    return SchoolDto(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      shortName: json['shortName'] ?? '',
      domain: json['domain'] ?? '',
      studentCodePattern: json['studentCodePattern'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'shortName': shortName,
      'domain': domain,
      'studentCodePattern': studentCodePattern,
    };
  }
}